### DISCUSSION TITLE
TIMEUTILS_SECONDS_1970_TO_2000

### DISCUSSION THREAD

**Discussion by iot49** [micropython/micropython#14298] [General]
"Some" micropython ports use a different epoch that starts in 2000, rather than 1970.

The difference is defined in `micropython/lib/shared/timeutils/timeutils.h` and set to either `TIMEUTILS_SECONDS_1970_TO_2000` or `0`, depending on the port.

Question: is it possible to get at this number from micropython code? 

Sending the output from `time.mktime(time.gmtime(0))` to a (unix-like) host and for comparison

```
time.mktime((2000, 1, 1, 0, 0, 0, 5, 1, -1)) - time.mktime(time.gmtime(0))
```

where `(2000, 1, 1, 0, 0, 0, 5, 1, -1)` is the output from `time.mktime(time.gmtime(0))` run on micropython (augmented by -1 as needed e.g. on the esp32) does the trick, but is convoluted. Hence the question for a simpler solution, preferably all contained within micropython.

**Comment by peterhinch** [comment]
You can determine the epoch with tests like
```python
if gmtime(0)[0] == 2000:
```
Time and date calculations are usually best done in seconds since the epoch. Conversion between epochs is accomplished by adding a constant. The easy way to determine the constant is in CPython:
```python
>>> from datetime import date
>>> (date(2000, 1, 1) - date(1900, 1, 1)).days * 24*60*60
3155673600
>>> (date(2000, 1, 1) - date(1970, 1, 1)).days * 24*60*60
946684800
>>> 
```

**Reply by iot49** [reply]
This works, thank you.
In micropython, I guess I just have to assume that the only two values used are `946684800` if `gmtime(0)[0] == 2000` and `0` otherwise.
